antarest/core/tasks/service.py

```diff
@@ -302,32 +303,36 @@ def list_db_tasks(self, task_filter: TaskListFilter, request_params: RequestPara
         return self.repo.list(task_filter, user)
 
     def await_task(self, task_id: str, timeout_sec: int = DEFAULT_AWAIT_MAX_TIMEOUT) -> None:
-        if task_id in self.tasks:
+        sanitized_task_id = base64.b64decode(base64.b64encode(task_id.encode("utf-8"))).decode("utf-8")
```

Comment:
This is not the right logic: the encode/decodes here only lead to having a sanitized ID identical to the input ID.

The rule for logs only targets end of lines, I think we should just have a sanity check which checks that the ID does not contain such characters.

The solution using base64 encoding as illustrated in sonar example should not be used when the input is correct, because it changes it (by definition, its aim is to obtain something where issues have been removed).
For example, base64 encoding of ID '7ac92d48-7be8-4961-a5ba7430d4b16fc' gives 'N2FjOTJkNDgtN2JlOC00OTYxLWE1YmE3NDMwZDRiMTZmY2Y='